src/VisualStudio/Core/Def/DebuggerIntelliSense/AbstractDebuggerIntelliSenseContext.cs

```diff
@@ -152,21 +148,10 @@ private bool TrySetContext(
             regionEdit.Apply();
         }
 
-        // Adjust the context point to ensure that the right information is in scope.
-        // For example, we may need to move the point to the end of the last statement in a method body
-        // in order to be able to access all local variables.
         var contextPoint = this.ContextBuffer.CurrentSnapshot.GetLineFromLineNumber(CurrentStatementSpan.iEndLine).Start + CurrentStatementSpan.iEndIndex;
-        var adjustedContextPoint = GetAdjustedContextPoint(contextPoint, document);
```

Comment:
I found having multiple abstract methods made this more confusing, and I needed to add in another concept which would have made this worse. Instead, just a single abstract method that just returns the end projection buffer and let the language implementations stitch them together themselves.